stix_shifter_modules/sumologic/stix_translation/query_constructor.py

```diff
@@ -148,6 +155,11 @@ def _parse_expression(self, expression, qualifier=None) -> str:
 
             comparison_string = self._parse_mapped_fields(self, expression, value, comparator, stix_field,
                                                           mapped_fields_array)
+
+            # IN operator logic
+            if expression.comparator == ComparisonComparators.In:
+                comparison_string = self._format_in(mapped_fields_array, expression.value)
```

Comment:
why do you need generate `comparison_string` twice?  is line 156/157 necessary if parsing IN operator? may be implement IN support inside `self._parse_mapped_fields()` or add a check so that you don't need to call self._parse_mapped_fields()